### ISSUE TITLE
is there a bug with 'relu'?

### ISSUE THREAD

**Issue by albertyou2** [fchollet/keras#3752]
Here is my network :

model = Sequential()
model.add(Convolution2D(128, 3, 3, border_mode='valid',input_shape=data.shape[-3:]))
model.add(Activation(act))
model.add(Convolution2D(128, 3, 3, border_mode='valid'))
model.add(Activation(act))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))
model.add(Convolution2D(256, 3, 3, border_mode='valid'))
model.add(Activation(act))
model.add(Convolution2D(256, 3, 3, border_mode='valid'))
model.add(Activation(act))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))
model.add(Convolution2D(512, 3, 3, border_mode='valid'))
model.add(Activation(act))
model.add(MaxPooling2D(pool_size=(2, 2)))
#
model.add(Flatten())
model.add(Dense(4096, init='normal'))
model.add(Activation(act))
model.add(Dropout(0.5))
model.add(Dense(4096, init='normal'))
model.add(Activation(act))
model.add(Dropout(0.5))

model.add(Dense(classNumber, init='normal'))
model.add(Activation('softmax'))

First I set The act = 'relu' ,  the loss and accuracy does't change.
Then I changed the act = 'tanh' and traing it again,  the loss and acc were normal。
I that a known issue？

**Comment by semitom** [comment]
Do you use negative inputs? ReLu will convert negative inputs to 0., whereas tanh can handle inputs between -1 and 1 just fine. ReLu is a very simple function, there won't be a bug in there.

**Comment by albertyou2** [comment]
@semitom  Thank you for reply me!

I think I found the reason :  Setting the momentum=0.9 for  SGD optimizer will cause this problem. I changed momentum to 0.5 the training is normal.Here is my optimizer and model setting:

sgd = SGD(lr=0.0005, decay=0.00001, momentum=0.9, nesterov=True)
model.compile(loss='categorical_crossentropy', optimizer=sgd,metrics=["accuracy"])
model.fit(data, label, batch_size=32, nb_epoch=3000,shuffle=True,verbose=1,validation_split=0.1)

But why did this happen?
I tried another Deep learning framework , let the momentum=0.9 ,using 'relu' activation and the same traing data .There was no problem.

Thank you again!
